### DISCUSSION TITLE
How to open dropdown (context menu) on right click?

### DISCUSSION THREAD

**Discussion by caleidon** [saadeghi/daisyui#757] [Q&A] [answered]
I need to be able to right click any image and open a context menu. I absolutely love the DaisyUI animations, and I'd like the dropdown to open and play that animation. However, I can't get this working because of several reasons.

1. As I am relatively new to CSS, I have no idea how the button actually triggers the dropdown to open. All I see is that they have some `tabindex`, and it doesn't make sense to me how it works. This directly ties into my next point.

2. The way I open my context menu is a bit specific. I use svelte, so when I right click, the entire `ContextMenu` component is actually created from scratch. This means I don't get the chance to click the button to trigger the animation. So, is it possible to immediately trigger the dropdown menu to open, as soon as the component is created? I know I can add `dropdown-open` to the class, but then it starts out fully opened and it doesn't trigger the beautiful animation.

**Comment by saadeghi** [comment, marked as answer]
dropdown works with focus. If you want a context menu, I think you should have a `menu`. make it hidden by default and make the position `absolute` then after right click, show the menu and set the position as the mouse position.

https://svelte.dev/repl/b79bb1755a894aca85aedd96419cf9ca?version=3.25.0

**Reply by caleidon** [reply]
I am aware of this method, but the problem is that I use a slightly different setup in which transition:fade doesn't work. Also, I don't want to use fade in the first place, but the beautiful, native animation of DaisyUI's dropdown, which doesn't work in this way. Is there some other workaround to achieve this?

**Reply by caleidon** [reply]
Nevermind. Got it to work by using bind:this={} to get the element, then calling .focus() to manually focus it and trigger the DaisyUI animation.
